packages/notifi-frontend-client/lib/client/ensureSource.ts

```diff
@@ -599,21 +599,29 @@ const getCustomFilterOptions = (
     case 'TOGGLE':
       return eventType.filterOptions;
     case 'HEALTH_CHECK': {
-      // Use synthetic ref values to get from input
+      // Use synthetic ref values to get from input (ratio)
       const healthRatioKey = `${eventType.name}__healthRatio`;
-      const healthRatio = resolveNumberRef(
-        healthRatioKey,
-        { type: 'ref', ref: healthRatioKey },
-        inputs,
-      );
-
+      const healthRatio =
+        resolveNumberRef(
+          healthRatioKey,
+          { type: 'ref', ref: healthRatioKey },
+          inputs,
+        ) ?? eventType.checkRatios[0].ratio; // Fallback to 1st checkRatios
```

Comment:
what if `checkRatios` is an empty array?

Reply:
Addressed! Add checkRatios length 0 case
```
if (!healthRatio || !thresholdDirection) {
    throw new Error('Failed to retrieve health ratio or direction');
}
```